osf/migrations/__init__.py

```diff
@@ -125,3 +125,60 @@ def update_permission_groups(sender, verbosity=0, **kwargs):
     if getattr(sender, 'label', None) == 'osf':
         update_admin_permissions(verbosity)
         update_provider_auth_groups(verbosity)
+
+
+class GroupHelper(object):
+    """ Helper for managing permission groups for a given provider during migrations.
+
+        The mixed-in functionality from ReviewProviderMixin is unavailable during migrations
+    """
+
+    def __init__(self, provider):
+        self.provider = provider
+
+    def format_group(self, name):
+        from osf.models.mixins import ReviewProviderMixin
+        if name not in ReviewProviderMixin.groups:
+            raise ValueError('Invalid reviews group: "{}"'.format(name))
+        return ReviewProviderMixin.group_format.format(self=self.provider, group=name)
+
+    def get_group(self, name):
+        from django.contrib.auth.models import Group
+        return Group.objects.get(name=self.format_group(name))
+
+    def update_provider_auth_groups(self):
+        from guardian.shortcuts import assign_perm, get_perms, remove_perm
+        from django.contrib.auth.models import Group
+
+        from osf.models.mixins import ReviewProviderMixin
+        for group_name, group_permissions in ReviewProviderMixin.groups.items():
+            group, created = Group.objects.get_or_create(name=self.format_group(group_name))
+            to_remove = set(get_perms(group, self.provider)).difference(group_permissions)
+            for p in to_remove:
+                remove_perm(p, group, self.provider)
+            for p in group_permissions:
+                assign_perm(p, group, self.provider)
+
+
+def populate_provider_notification_subscriptions_registration(app=None, state=None):
```

Comment:
make post migration signal, remove preprintprovider legacy migration and tinker more